src/Sentry/Extensibility/SentryStackTraceFactory.cs

```diff
@@ -36,287 +21,18 @@ public class SentryStackTraceFactory : ISentryStackTraceFactory
-    /// names that appears in the Sentry UI will match the function and module
-    /// names in the original source-code.
-    /// </para>
-    private static void DemangleAsyncFunctionName(SentryStackFrame frame)
-    {
-        if (frame.Module == null || frame.Function != "MoveNext")
-        {
-            return;
-        }
-
-        //  Search for the function name in angle brackets followed by d__<digits>.
-        //
-        // Change:
-        //   RemotePrinterService+<UpdateNotification>d__24 in MoveNext at line 457:13
-        // to:
-        //   RemotePrinterService in UpdateNotification at line 457:13
-
-        var match = RegexAsyncFunctionName.Match(frame.Module);
-        if (match.Success && match.Groups.Count == 3)
-        {
-            frame.Module = match.Groups[1].Value;
-            frame.Function = match.Groups[2].Value;
-        }
-    }
-
-    /// <summary>
-    /// Clean up function names for anonymous lambda calls.
-    /// </summary>
-    internal static void DemangleAnonymousFunction(SentryStackFrame frame)
-    {
-        if (frame.Function == null)
-        {
-            return;
-        }
-
-        // Search for the function name in angle brackets followed by b__<digits/letters>.
-        //
-        // Change:
-        //   <BeginInvokeAsynchronousActionMethod>b__36
-        // to:
-        //   BeginInvokeAsynchronousActionMethod { <lambda> }
-
-        var match = RegexAnonymousFunction.Match(frame.Function);
-        if (match.Success && match.Groups.Count == 2)
-        {
-            frame.Function = match.Groups[1].Value + " { <lambda> }";
-        }
-    }
-
-    /// <summary>
-    /// Remove return type from module in a Task with a Lambda with a return value.
-    /// This was seen in Unity, see https://github.com/getsentry/sentry-unity/issues/845
-    /// </summary>
-    internal static void DemangleLambdaReturnType(SentryStackFrame frame)
-    {
-        if (frame.Module == null)
-        {
-            return;
-        }
-
-        // Change:
-        //   System.Threading.Tasks.Task`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]] in InnerInvoke
-        //   or System.Collections.Generic.List`1[[UnityEngine.Events.PersistentCall, UnityEngine.CoreModule, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]] in get_Item
-        // to:
-        //   System.Threading.Tasks.Task`1 in InnerInvoke`
-        //   or System.Collections.Generic.List`1 in get_Item
-        var match = RegexAsyncReturn.Match(frame.Module);
-        if (match.Success && match.Groups.Count == 2)
-        {
-            frame.Module = match.Groups[1].Value;
-        }
+        var stackTrace = exception is null ? new StackTrace(true) : new StackTrace(exception, true);
+        var result = DebugStackTrace.Create(_options, stackTrace, isCurrentStackTrace);
+        _options.LogDebug($"Created {typeof(DebugStackTrace).Name} with {0} frames.", result.Frames.Count);
```

Comment:
The goal was to avoid allocations if Debug is turned off (the default behaviour).

That's why originally all calls were to a nullable reference (`_options.Logger?.Log(..)`) so if `Debug=false`, not even a call was done.
Now there's an extension method so the call is going to be made, we can avoid the interpolation at least

```suggestion
        _options.LogDebug("Created {0} with {1} frames.", typeof(DebugStackTrace).Name, result.Frames.Count);
```